scanner/src/main/kotlin/scanners/scanoss/ScanOss.kt

```diff
@@ -37,14 +37,15 @@ import org.ossreviewtoolkit.scanner.PathScanner
 import org.ossreviewtoolkit.utils.common.safeDeleteRecursively
 import org.ossreviewtoolkit.utils.core.createOrtTempDir
 
-class ScanOss(
+class ScanOss internal constructor(
```

Comment:
Why make the constructor internal? So far, `FossId` is the only other scanner with an internal constructor. And I tend to rather make it public there, too.

Reply:
Isn't it because you should not instantiate the scanner directly and go through the factory instead ?
So I understand the design of Oliver (who did it for FossID).

Reply:
Well, ORT itself should use the factory (except maybe in tests), yes. But I don't see a reason why people who use ORT's libraries programmatically (if there are any 😀) instantiate the class directly.

If we make this constructor internal, it would be internal for all scanner implementations, and even going father, for all ORT classes that have factories.